packages/react/src/auto/polaris/PolarisAutoTable.tsx

```diff
@@ -93,13 +95,24 @@ export const PolarisAutoTable = <
         onQueryChange={search.set}
         onQueryClear={search.clear}
       />
+
+      {error && (
+        <Box paddingBlockStart="200" paddingBlockEnd="1000">
+          <Banner title={error.message} tone="critical" />
+        </Box>
+      )}
+
       <IndexTable
         {...polarisTableProps}
         resourceName={resourceName}
         emptyState={<EmptySearchResult title={`No ${resourceName.plural} yet`} description={""} withIllustration />}
         loading={fetching}
         hasMoreItems={page.hasNextPage}
-        itemCount={rows?.length ?? 0}
+        itemCount={
+          error
+            ? 1 // Don't show the empty state if there's an error
+            : rows?.length ?? 0
+        }
```

Comment:
This really just adds the disabled pagination buttons when we are errored. What do you think about something like 
```
error ? <ErrorBanner/> : <IndexTable/>
```
so that error behavior doesn't need to be added to the index table?

Reply:
It's not entirely the same because with your code, it will look like this:
<img width="646" alt="CleanShot 2024-07-19 at 09 47 23@2x" src="https://github.com/user-attachments/assets/a983e82a-fc75-4689-af84-4c0b5261a862">

It doesn't feel like it's inside the table.